shared-libs/transitions/src/db.js

```diff
@@ -0,0 +1,39 @@
+const logger = require('./lib/logger'),
+      _ = require('underscore');
+const { UNIT_TEST_ENV } = process.env;
+
+if (UNIT_TEST_ENV) {
+  const stubMe = functionName => () => {
+    logger.error(
+      new Error(
+        `db.${functionName}() not stubbed!  UNIT_TEST_ENV=${UNIT_TEST_ENV}.  Please stub PouchDB functions that will be interacted with in unit tests.`
+      )
+    );
+    process.exit(1);
+  };
+
+  module.exports.medic = {
+    allDocs: stubMe('allDocs'),
+    bulkDocs: stubMe('bulkDocs'),
+    put: stubMe('put'),
+    remove: stubMe('remove'),
+    post: stubMe('post'),
+    query: stubMe('query'),
+    get: stubMe('get'),
+    changes: stubMe('changes'),
+  };
+
+  module.exports.sentinel = {
+    allDocs: stubMe('allDocs'),
+    bulkDocs: stubMe('bulkDocs'),
+    put: stubMe('put'),
+    post: stubMe('post'),
+    query: stubMe('query'),
+    get: stubMe('get'),
+    changes: stubMe('changes'),
+  };
+}
+
+module.exports.init = db => {
+  Object.assign(module.exports, _.omit(db, 'init'));
+};
```

Comment:
Is this solely for testing? If so, can we move it into the tests dir for clarity?

Reply:
Well, not for testing. 
https://github.com/medic/medic/blob/4857-detached-sentinel-bis/shared-libs/transitions/src/index.js#L5 

I'm passing the whole `db` and whole `logger` objects to this library so it simulates functioning in the environment that it was called from. 

The DB-s, for example, have headers attached to the `fetch` functions (`X-Medic-Service: api`, `X-Medic-User: sentinel`) for haproxy auditing. 
I thought about just passing these specific headers and duplicate the code to initialize the `db` but went with this instead. 
I'll update them to just pass the headers and other specific info needed.

Reply:
Why not just use the sourceDb directly instead of shallow cloning it?